NeoForge 26.1.2.21 broke compatibility

### Describe the bug

<img width="851" height="507" alt="Image" src="https://github.com/user-attachments/assets/63951c2c-4f28-42c2-a136-085bedc95fda" />

Whenever you try to open the game (I used NeoForge both 26.1.2.28-beta and 26.1.2.29-beta), it throws this error

### Isolated reproduction

Yes

### Minecraft version

26.1.2

### Loader used

NeoForge

### Loader version (Fabric/Forge/NeoForge)

26.1.2.28-beta

### Mod version

26.1.2.2

### Minecraft environment

Single player

### Configuration

```json
I did not edited anything because I can't
```

### Relevant log output

```shell
Prism Launcher version: 11.0.2 (official)

Launched instance in online mode

login.microsoftonline.com resolves to:
  2603:1026:3000:150::8, 2603:1026:3000:148::e, 2603:1027:1:158::c, 2603:1026:3000:158::3, 2603:1026:3000:148::6, 2603:1026:3000:150::a, 2603:1027:1:158::a, 2603:1027:1:148::4, 40.126.32.133, 40.126.32.134, 20.190.160.2, 20.190.160.132, 40.126.32.138, 20.190.160.64, 20.190.160.5, 40.126.32.68

session.minecraft.net resolves to:
  2603:1061:14:c0::1, 150.171.109.193

textures.minecraft.net resolves to:
  2603:1061:14:c0::1, 150.171.109.193

api.mojang.com resolves to:
  2603:1061:14:c0::1, 150.171.109.193

Minecraft folder is:
  /Users/********/Library/Application Support/PrismLauncher/instances/26.1.2(1)/minecraft

Compatible Java found at: /Users/********/Library/Application Support/PrismLauncher/java/java-runtime-epsilon/bin/java.
Java path is:
  /Users/********/Library/Application Support/PrismLauncher/java/java-runtime-epsilon/bin/java
Java is version 25.0.1, using 64 (aarch64) architecture, from Microsoft

OS: macOS Tahoe (26.4.1) | darwin | 25.4.0
CPU: Apple M5
RAM: 16384 MiB (available: 4520 MiB)
OpenGL driver vendor: Apple
OpenGL renderer: Apple M5
OpenGL driver version: 2.1 Metal - 90.5

Launcher: standard
Main class: io.github.zekerzhayard.forgewrapper.installer.Main

Mods:
  [✔] FallingTree-26.1.2-25

Traits:
  XR:Initial
  feature:is_quick_play_singleplayer
  feature:is_quick_play_multiplayer
  FirstThreadOnMacOS

Libraries:
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-freetype-natives-macos-arm64/3.4.1/lwjgl-freetype-natives-macos-arm64-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-freetype/3.4.1/lwjgl-freetype-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-glfw-natives-macos-arm64/3.4.1/lwjgl-glfw-natives-macos-arm64-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-glfw/3.4.1/lwjgl-glfw-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-jemalloc-natives-macos-arm64/3.4.1/lwjgl-jemalloc-natives-macos-arm64-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-jemalloc/3.4.1/lwjgl-jemalloc-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-natives-macos-arm64/3.4.1/lwjgl-natives-macos-arm64-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-openal-natives-macos-arm64/3.4.1/lwjgl-openal-natives-macos-arm64-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-openal/3.4.1/lwjgl-openal-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-opengl-natives-macos-arm64/3.4.1/lwjgl-opengl-natives-macos-arm64-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-opengl/3.4.1/lwjgl-opengl-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-shaderc-natives-macos-arm64/3.4.1/lwjgl-shaderc-natives-macos-arm64-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-shaderc/3.4.1/lwjgl-shaderc-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-spvc-natives-macos-arm64/3.4.1/lwjgl-spvc-natives-macos-arm64-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-spvc/3.4.1/lwjgl-spvc-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-stb-natives-macos-arm64/3.4.1/lwjgl-stb-natives-macos-arm64-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-stb/3.4.1/lwjgl-stb-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-tinyfd-natives-macos-arm64/3.4.1/lwjgl-tinyfd-natives-macos-arm64-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-tinyfd/3.4.1/lwjgl-tinyfd-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-vma-natives-macos-arm64/3.4.1/lwjgl-vma-natives-macos-arm64-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-vma/3.4.1/lwjgl-vma-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-vulkan-natives-macos-arm64/3.4.1/lwjgl-vulkan-natives-macos-arm64-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl-vulkan/3.4.1/lwjgl-vulkan-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/lwjgl/lwjgl/3.4.1/lwjgl-3.4.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/at/yawk/lz4/lz4-java/1.10.1/lz4-java-1.10.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/ca/weblite/java-objc-bridge/1.1/java-objc-bridge-1.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/com/azure/azure-json/1.4.0/azure-json-1.4.0.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/com/github/oshi/oshi-core/6.9.0/oshi-core-6.9.0.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/com/google/code/gson/gson/2.13.2/gson-2.13.2.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/com/google/guava/failureaccess/1.0.3/failureaccess-1.0.3.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/com/google/guava/guava/33.5.0-jre/guava-33.5.0-jre.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/com/ibm/icu/icu4j/77.1/icu4j-77.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/com/microsoft/azure/msal4j/1.23.1/msal4j-1.23.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/com/mojang/authlib/7.0.63/authlib-7.0.63.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/com/mojang/blocklist/1.0.10/blocklist-1.0.10.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/com/mojang/brigadier/1.3.10/brigadier-1.3.10.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/com/mojang/datafixerupper/9.0.19/datafixerupper-9.0.19.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/com/mojang/jtracy/1.0.37/jtracy-1.0.37.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/com/mojang/logging/1.6.11/logging-1.6.11.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/com/mojang/patchy/2.2.10/patchy-2.2.10.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/com/mojang/text2speech/1.18.11/text2speech-1.18.11.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/commons-codec/commons-codec/1.19.0/commons-codec-1.19.0.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/commons-io/commons-io/2.20.0/commons-io-2.20.0.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/io/netty/netty-buffer/4.2.7.Final/netty-buffer-4.2.7.Final.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/io/netty/netty-codec-base/4.2.7.Final/netty-codec-base-4.2.7.Final.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/io/netty/netty-codec-compression/4.2.7.Final/netty-codec-compression-4.2.7.Final.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/io/netty/netty-codec-http/4.2.7.Final/netty-codec-http-4.2.7.Final.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/io/netty/netty-common/4.2.7.Final/netty-common-4.2.7.Final.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/io/netty/netty-handler/4.2.7.Final/netty-handler-4.2.7.Final.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/io/netty/netty-resolver/4.2.7.Final/netty-resolver-4.2.7.Final.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/io/netty/netty-transport-classes-epoll/4.2.7.Final/netty-transport-classes-epoll-4.2.7.Final.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/io/netty/netty-transport-classes-kqueue/4.2.7.Final/netty-transport-classes-kqueue-4.2.7.Final.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/io/netty/netty-transport-native-unix-common/4.2.7.Final/netty-transport-native-unix-common-4.2.7.Final.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/io/netty/netty-transport/4.2.7.Final/netty-transport-4.2.7.Final.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/it/unimi/dsi/fastutil/8.5.18/fastutil-8.5.18.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/net/java/dev/jna/jna-platform/5.17.0/jna-platform-5.17.0.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/net/java/dev/jna/jna/5.17.0/jna-5.17.0.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/net/sf/jopt-simple/jopt-simple/5.0.4/jopt-simple-5.0.4.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/apache/commons/commons-compress/1.28.0/commons-compress-1.28.0.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/apache/commons/commons-lang3/3.19.0/commons-lang3-3.19.0.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/apache/logging/log4j/log4j-api/2.25.2/log4j-api-2.25.2.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/apache/logging/log4j/log4j-core/2.25.2/log4j-core-2.25.2.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/apache/logging/log4j/log4j-slf4j2-impl/2.25.2/log4j-slf4j2-impl-2.25.2.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/jcraft/jorbis/0.0.17/jorbis-0.0.17.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/joml/joml/1.10.8/joml-1.10.8.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/jspecify/jspecify/1.0.0/jspecify-1.0.0.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/slf4j/slf4j-api/2.0.17/slf4j-api-2.0.17.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/io/github/zekerzhayard/ForgeWrapper/prism-2025-12-07/ForgeWrapper-prism-2025-12-07.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/net/neoforged/fancymodloader/earlydisplay/11.0.13/earlydisplay-11.0.13.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/net/neoforged/fancymodloader/loader/11.0.13/loader-11.0.13.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/net/neoforged/accesstransformers/11.0.2/accesstransformers-11.0.2.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/ow2/asm/asm-commons/9.9.1/asm-commons-9.9.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/net/neoforged/mergetool/2.0.7/mergetool-2.0.7-api.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/ow2/asm/asm-util/9.9.1/asm-util-9.9.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/ow2/asm/asm-analysis/9.9.1/asm-analysis-9.9.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/ow2/asm/asm-tree/9.9.1/asm-tree-9.9.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/net/neoforged/bus/8.0.5/bus-8.0.5.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/ow2/asm/asm/9.9.1/asm-9.9.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/com/electronwill/night-config/toml/3.8.3/toml-3.8.3.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/com/electronwill/night-config/core/3.8.3/core-3.8.3.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/net/neoforged/JarJarSelector/0.5.0/JarJarSelector-0.5.0.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/net/neoforged/JarJarMetadata/0.5.0/JarJarMetadata-0.5.0.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/apache/maven/maven-artifact/3.9.9/maven-artifact-3.9.9.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/net/jodah/typetools/0.6.3/typetools-0.6.3.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/net/minecrell/terminalconsoleappender/1.3.0/terminalconsoleappender-1.3.0.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/net/fabricmc/sponge-mixin/0.17.3+mixin.0.8.7/sponge-mixin-0.17.3+mixin.0.8.7.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/net/neoforged/accesstransformers/at-parser/11.0.2/at-parser-11.0.2.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/jline/jline-reader/3.20.0/jline-reader-3.20.0.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/jline/jline-terminal/3.20.0/jline-terminal-3.20.0.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/net/neoforged/srgutils/1.0.10/srgutils-1.0.10.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/com/google/guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/com/google/errorprone/error_prone_annotations/2.41.0/error_prone_annotations-2.41.0.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/com/google/j2objc/j2objc-annotations/3.1/j2objc-annotations-3.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/org/codehaus/plexus/plexus-utils/3.5.1/plexus-utils-3.5.1.jar
  /Users/********/Library/Application Support/PrismLauncher/libraries/com/mojang/minecraft/26.1.2/minecraft-26.1.2-client.jar

Native libraries:

Natives path:
  /Users/********/Library/Application Support/PrismLauncher/instances/26.1.2(1)/natives

Minecraft arguments:
  --username  --version 26.1.2 --gameDir /Users/********/Library/Application Support/PrismLauncher/instances/26.1.2(1)/minecraft --assetsDir /Users/********/Library/Application Support/PrismLauncher/assets --assetIndex 30 --uuid  --accessToken  --userType  --versionType release --fml.neoForgeVersion 26.1.2.28-beta --fml.mcVersion 26.1.2 --fml.neoFormVersion 1

Window size: 854 x 480

Java arguments:
  -Duser.language=en -Xdock:icon=icon.png -Xdock:name="Prism Launcher: 26.1.2" -XstartOnFirstThread -Xms512m -Xmx4096m

Minecraft process ID: 46959


WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::staticFieldBase has been called by io.github.zekerzhayard.forgewrapper.installer.util.ModuleUtil (file:/Users/********/Library/Application%20Support/PrismLauncher/libraries/io/github/zekerzhayard/ForgeWrapper/prism-2025-12-07/ForgeWrapper-prism-2025-12-07.jar)
WARNING: Please consider reporting this to the maintainers of class io.github.zekerzhayard.forgewrapper.installer.util.ModuleUtil
WARNING: sun.misc.Unsafe::staticFieldBase will be removed in a future release
JVM info: Microsoft - 25.0.1 - 25.0.1+8-LTS
java.net.preferIPv4Stack=true
Current Time: 25/04/2026 18:23:53
Created Temporary Directory: /var/folders/q1/0bf4cd416rdc06zc3y0drj3h0000gn/T/neoforge_installer9877247322793824175
  Extracting: /data/client.lzma
Building Processor
===============================================================================
Processor: net.neoforged.installertools:installertools -> PROCESS_MINECRAFT_JAR
  Cache: 
    /Users/********/Library/Application Support/PrismLauncher/libraries/net/neoforged/minecraft-client-patched/26.1.2.28-beta/minecraft-client-patched-26.1.2.28-beta.jar Validated: 3b80d83b016f22a4a6927f48ef3960eeed75f390
  Cache Hit!
2026-04-25T16:23:53.651191Z main WARN Advanced terminal features are not available in this environment
[18:23:53] [main/INFO] [ne.ne.fm.st.Entrypoint/]: JVM Uptime at startup: 548ms
[18:23:53] [main/INFO] [ne.ne.fm.lo.FMLLoader/]: Starting FancyModLoader version 11.0.13 (CLIENT in PROD)
[18:23:53] [main/INFO] [ne.ne.fm.lo.FMLLoader/]: Game directory: /Users/********/Library/Application Support/PrismLauncher/instances/26.1.2(1)/minecraft
[18:23:54] [main/INFO] [ne.ne.fm.lo.EarlyServiceDiscovery/]: Found 2 early service jars (out of 96) in 35ms
[18:23:54] [main/INFO] [ne.ne.fm.lo.FMLLoader/]: Loading FML Early Services:
[18:23:54] [main/INFO] [ne.ne.fm.lo.FMLLoader/]:  - /Users/********/Library/Application Support/PrismLauncher/libraries/net/neoforged/fancymodloader/earlydisplay/11.0.13/earlydisplay-11.0.13.jar
[18:23:54] [main/INFO] [ne.ne.fm.lo.FMLLoader/]:  - /Users/********/Library/Application Support/PrismLauncher/libraries/net/neoforged/fancymodloader/loader/11.0.13/loader-11.0.13.jar
[18:23:54] [main/INFO] [ne.ne.fm.lo.ImmediateWindowHandler/]: Loading ImmediateWindowProvider fmlearlywindow
WARNING: A restricted method in java.lang.System has been called
WARNING: java.lang.System::load has been called by org.lwjgl.system.Library$$Lambda/0x000003c00028cd00 in an unnamed module (file:/Users/********/Library/Application%20Support/PrismLauncher/libraries/org/lwjgl/lwjgl/3.4.1/lwjgl-3.4.1.jar)
WARNING: Use --enable-native-access=ALL-UNNAMED to avoid a warning for callers in this module
WARNING: Restricted methods will be blocked in a future release unless native access is enabled
[18:23:54] [main/WARN] [EARLYDISPLAY/]: Failed to set window icon: [0x1000C] Cocoa: Regular windows do not have icons on macOS
[18:23:54] [ForkJoinPool.commonPool-worker-6/INFO] [ne.ne.fm.lo.mo.lo.GameLocator/]: Marking unmodified client jar as claimed to prevent loading: /Users/********/Library/Application Support/PrismLauncher/libraries/com/mojang/minecraft/26.1.2/minecraft-26.1.2-client.jar
[18:23:54] [ForkJoinPool.commonPool-worker-6/INFO] [ne.ne.fm.lo.mo.lo.GameLocator/]: Failed to find common Minecraft classes and resources on the classpath. Assuming we're launching production.
[18:23:54] [fml-loadingscreen/INFO] [ne.ne.fm.ea.re.LoadingScreenRenderer/]: GL info: Apple M5 GL version 4.1 Metal - 90.5, Apple
[18:23:54] [ForkJoinPool.commonPool-worker-6/ERROR] [ne.ne.fm.lo.mo.ModFile/LOADING]: Access transformer file META-INF/neoforge.accesstransformer.cfg provided by mod fallingtree does not exist!
[18:23:54] [ForkJoinPool.commonPool-worker-6/INFO] [ne.ne.fm.lo.mo.lo.JarInJarDependencyLocator/]: Found 2 dependencies adding them to mods collection
[18:23:54] [ForkJoinPool.commonPool-worker-6/INFO] [ne.ne.fm.lo.mo.ModDiscoverer/]: 
     Mod List:
		Name Version (Mod Id)
		FallingTree 25 (fallingtree)
		Minecraft 26.1.2 (minecraft)
		NeoForge 26.1.2.28-beta (neoforge)
[18:23:54] [main/INFO] [mixin/]: SpongePowered MIXIN Subsystem Version=0.8.7 Source=file:/Users/********/Library/Application%20Support/PrismLauncher/libraries/net/fabricmc/sponge-mixin/0.17.3+mixin.0.8.7/sponge-mixin-0.17.3+mixin.0.8.7.jar Service=FML Env=CLIENT
[18:23:54] [main/INFO] [ne.ne.fm.lo.FMLLoader/]: Loading FML Plugins:
[18:23:54] [main/INFO] [ne.ne.fm.lo.FMLLoader/]:  - ~libraries/net/neoforged/neoforge/26.1.2.28-beta/neoforge-26.1.2.28-beta-universal.jar > net.neoforged.neoforge-coremods-26.1.2.28-beta.jar
[18:23:55] [main/INFO] [ne.ne.fm.lo.FMLLoader/]: Building game content classloader:
 - fallingtree (jar(mods/FallingTree-26.1.2-25.jar))
 - minecraft (jar(~libraries/net/neoforged/minecraft-client-patched/26.1.2.28-beta/minecraft-client-patched-26.1.2.28-beta.jar))
 - mixinextras.neoforge (jar(~libraries/net/neoforged/neoforge/26.1.2.28-beta/neoforge-26.1.2.28-beta-universal.jar > mixinextras-neoforge-0.5.4.jar))
 - neoforge (jar(~libraries/net/neoforged/neoforge/26.1.2.28-beta/neoforge-26.1.2.28-beta-universal.jar))
 - net.neoforged.fml.generated (empty(VirtualJar/net.neoforged.fml.generated))
[18:23:55] [main/INFO] [ne.ne.fm.lo.FMLLoader/]: Built game content classloader in 16ms
[18:23:55] [Datafixer Bootstrap/INFO] [mojang/DataFixerBuilder]: 294 Datafixer optimizations took 210 milliseconds
[18:23:56] [pool-8-thread-1/INFO] [MixinExtras|Service/]: Initializing MixinExtras via com.llamalad7.mixinextras.service.MixinExtrasServiceImpl(version=0.5.4).
[18:23:59] [modloading-worker-0/ERROR] [ne.ne.fm.ja.FMLModContainer/LOADING]: Failed to create mod instance. ModID: fallingtree, class fr.rakambda.fallingtree.neoforge.FallingTree
java.lang.NoClassDefFoundError: net/neoforged/neoforge/event/level/BlockEvent$BreakEvent
	at java.base/java.lang.ClassLoader.defineClass1(Native Method) ~[?:?] {}
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:962) ~[?:?] {}
	at net.neoforged.fml.classloading.ModuleClassLoader.readerToClass(ModuleClassLoader.java:229) ~[loader-11.0.13.jar:11.0] {}
	at net.neoforged.fml.classloading.ModuleClassLoader.loadClass(ModuleClassLoader.java:245) ~[loader-11.0.13.jar:11.0] {}
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:490) ~[?:?] {}
	at TRANSFORMER/fallingtree@25/fr.rakambda.fallingtree.neoforge.FallingTree.<init>(FallingTree.java:20) ~[FallingTree-26.1.2-25.jar:?] {}
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62) ~[?:?] {}
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499) ~[?:?] {}
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:483) ~[?:?] {}
	at net.neoforged.fml.javafmlmod.FMLModContainer.constructMod(FMLModContainer.java:118) [loader-11.0.13.jar:11.0] {}
	at net.neoforged.fml.ModLoader.lambda$constructMods$0(ModLoader.java:125) [loader-11.0.13.jar:11.0] {}
	at net.neoforged.fml.ModLoader.lambda$dispatchParallelTask$3(ModLoader.java:203) [loader-11.0.13.jar:11.0] {}
	at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:955) [?:?] {}
	at java.base/java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:932) [?:?] {}
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:504) [?:?] {}
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:511) [?:?] {}
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1450) [?:?] {}
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2019) [?:?] {}
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:187) [?:?] {}
Caused by: java.lang.ClassNotFoundException: net.neoforged.neoforge.event.level.BlockEvent$BreakEvent
	at net.neoforged.fml.classloading.ModuleClassLoader.loadClass(ModuleClassLoader.java:254) ~[loader-11.0.13.jar:11.0] {}
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:490) ~[?:?] {}
	... 19 more
[18:23:59] [modloading-worker-0/INFO] [ne.ne.ne.co.NeoForgeMod/NEOFORGE-MOD]: NeoForge mod loading, version 26.1.2.28-beta, for MC 26.1.2
[18:23:59] [main/FATAL] [ne.ne.fm.ModLoader/LOADING]: Failed to wait for future Mod Construction, 1 errors found
[18:23:59] [main/FATAL] [ne.ne.ne.co.NeoForgeMod/]: Preparing crash report with UUID e45e96ae-0060-484a-b1a8-a216d57a15bb
---- Minecraft Crash Report ----
// You're mean.
Time: 2026-04-25 18:23:59
Description: Bootstrap
net.neoforged.fml.ModLoadingException: Loading errors encountered:
	- FallingTree (fallingtree) has failed to load correctly
	  java.lang.NoClassDefFoundError: net/neoforged/neoforge/event/level/BlockEvent$BreakEvent
	at net.neoforged.fml.ModLoader.waitForFuture(ModLoader.java:253) ~[loader-11.0.13.jar:11.0] {}
	at net.neoforged.fml.ModLoader.dispatchParallelTask(ModLoader.java:216) ~[loader-11.0.13.jar:11.0] {}
	at net.neoforged.fml.ModLoader.constructMods(ModLoader.java:124) ~[loader-11.0.13.jar:11.0] {}
	at net.neoforged.fml.ModLoader.gatherAndInitializeMods(ModLoader.java:101) ~[loader-11.0.13.jar:11.0] {}
	at TRANSFORMER/neoforge@26.1.2.28-beta/net.neoforged.neoforge.internal.CommonModLoader.begin(CommonModLoader.java:49) ~[neoforge-26.1.2.28-beta-universal.jar:?] {}
	at TRANSFORMER/neoforge@26.1.2.28-beta/net.neoforged.neoforge.client.loading.ClientModLoader.begin(ClientModLoader.java:55) ~[neoforge-26.1.2.28-beta-universal.jar:?] {}
	at TRANSFORMER/minecraft@26.1.2/net.minecraft.client.main.Main.main(Main.java:122) [minecraft-client-patched-26.1.2.28-beta.jar:?] {}
	at net.neoforged.fml.startup.Client.main(Client.java:19) [loader-11.0.13.jar:11.0] {}
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104) ~[?:?] {}
	at java.base/java.lang.reflect.Method.invoke(Method.java:565) ~[?:?] {}
	at io.github.zekerzhayard.forgewrapper.installer.Main.main(Main.java:67) [ForgeWrapper-prism-2025-12-07.jar:prism-2025-12-07] {}
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:115) [NewLaunch.jar:?] {}
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129) [NewLaunch.jar:?] {}
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70) [NewLaunch.jar:?] {}
A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------
-- Head --
Thread: main
Stacktrace:
	at net.neoforged.fml.ModLoader.waitForFuture(ModLoader.java:253) ~[loader-11.0.13.jar:11.0] {}
	at net.neoforged.fml.ModLoader.dispatchParallelTask(ModLoader.java:216) ~[loader-11.0.13.jar:11.0] {}
	at net.neoforged.fml.ModLoader.constructMods(ModLoader.java:124) ~[loader-11.0.13.jar:11.0] {}
	at net.neoforged.fml.ModLoader.gatherAndInitializeMods(ModLoader.java:101) ~[loader-11.0.13.jar:11.0] {}
	at TRANSFORMER/neoforge@26.1.2.28-beta/net.neoforged.neoforge.internal.CommonModLoader.begin(CommonModLoader.java:49) ~[neoforge-26.1.2.28-beta-universal.jar:?] {}
	at TRANSFORMER/neoforge@26.1.2.28-beta/net.neoforged.neoforge.client.loading.ClientModLoader.begin(ClientModLoader.java:55) ~[neoforge-26.1.2.28-beta-universal.jar:?] {}
-- Initialization --
Details:
	Modules: 
Stacktrace:
	at TRANSFORMER/minecraft@26.1.2/net.minecraft.client.main.Main.main(Main.java:122) [minecraft-client-patched-26.1.2.28-beta.jar:?] {}
	at net.neoforged.fml.startup.Client.main(Client.java:19) [loader-11.0.13.jar:11.0] {}
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104) ~[?:?] {}
	at java.base/java.lang.reflect.Method.invoke(Method.java:565) ~[?:?] {}
	at io.github.zekerzhayard.forgewrapper.installer.Main.main(Main.java:67) [ForgeWrapper-prism-2025-12-07.jar:prism-2025-12-07] {}
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:115) [NewLaunch.jar:?] {}
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129) [NewLaunch.jar:?] {}
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70) [NewLaunch.jar:?] {}
-- System Details --
Details:
	Minecraft Version: 26.1.2
	Minecraft Version ID: 26.1.2
	Operating System: Mac OS X (aarch64) version 26.4.1
	Java Version: 25.0.1, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 230364336 bytes (219 MiB) / 536870912 bytes (512 MiB) up to 4294967296 bytes (4096 MiB)
	Memory (heap): init: 512MiB, used: 290MiB, committed: 512MiB, max: 4096MiB
	Memory (non-head): init: 007MiB, used: 126MiB, committed: 128MiB, max: 000MiB
	CPUs: 10
	Processor Vendor: Apple Inc.
	Processor Name: Apple M5
	Identifier: Apple Inc. Family 0x1b588bb3 Model 0 Stepping 0
	Microarchitecture: ARM64 SoC: Firestorm + Icestorm
	Frequency (GHz): 0.00
	Number of physical packages: 1
	Number of physical CPUs: 10
	Number of logical CPUs: 10
	Graphics card #0 name: Apple M5
	Graphics card #0 vendor: Apple (0x106b)
	Graphics card #0 VRAM (MiB): 0.00
	Graphics card #0 deviceId: unknown
	Graphics card #0 versionInfo: unknown
	Memory slot #0 capacity (MiB): 0.00
	Memory slot #0 clockSpeed (GHz): 0.00
	Memory slot #0 type: unknown
	Virtual memory max (MiB): 17408.00
	Virtual memory used (MiB): 12460.91
	Swap memory total (MiB): 1024.00
	Swap memory used (MiB): 369.88
	Space in storage for jna.tmpdir (MiB): <path not set>
	Space in storage for org.lwjgl.system.SharedLibraryExtractPath (MiB): <path not set>
	Space in storage for io.netty.native.workdir (MiB): <path not set>
	Space in storage for java.io.tmpdir (MiB): available: 177884.56, total: 471432.09
	Space in storage for workdir (MiB): available: 177884.56, total: 471432.09
	JVM Flags: 2 total; -Xms512m -Xmx4096m
	Debug Flags: 0 total; 
	Launched Version: 26.1.2
	Launcher name: PrismLauncher
	Backend library: LWJGL version 3.4.1+2
	Backend API: Unknown
	Window size: <not initialized>
	GFLW Platform: cocoa
	Render Extensions: <no renderer available>
	GL debug messages: <no renderer available>
	Is Modded: Definitely; Client brand changed to 'neoforge'
	Universe: 404
	Type: Client
	Locale: en_US
	System encoding: UTF-8
	File encoding: UTF-8
	CPU: 10x Apple M5
	Class Processors: 
		00.00%: neoforge:computing_frames (marker)
		00.00%: neoforge:runtime_enum_extender
		04.95%: neoforge:access_transformer
		00.09%: neoforge:mixin
		00.00%: neoforge:simple_processors_default (marker)
		00.02%: neoforge.coremods:field_to_getter.net.minecraft.world.level.biome.biome
		00.02%: neoforge.coremods:field_to_getter.net.minecraft.world.level.block.flowerpotblock
		00.02%: neoforge.coremods:field_to_getter.net.minecraft.world.level.levelgen.structure.structure
		00.39%: neoforge.coremods:method_redirector
	Mod List: 
		FallingTree-26.1.2-25.jar                         |FallingTree                   |fallingtree                   |25                  |Manifest: 000c07efd4d210802f20f1663f41abc8736a3b3d1b07d563bc14504558fbd496
		minecraft-client-patched-26.1.2.28-beta.jar       |Minecraft                     |minecraft                     |26.1.2              |Manifest: da9087ccf0d2fdd6569d9acc2595acae24bc150562036ff769e43383d61eb8f7
		neoforge-26.1.2.28-beta-universal.jar             |NeoForge                      |neoforge                      |26.1.2.28-beta      |Manifest: 631aab87d81a3715c414c9c9af661bc3dde110ace14cb8b2d8d7f5cbed713a48
	Crash Report UUID: e45e96ae-0060-484a-b1a8-a216d57a15bb
	FML: 11.0.13
	NeoForge: 26.1.2.28-beta
[18:23:59] [main/FATAL] [ne.ne.ne.co.NeoForgeMod/]: Preparing crash report with UUID 8dabe806-bc44-4465-b6c7-c07baafdbaea
#@!@# Game crashed! Crash report saved to: #@!@# /Users/********/Library/Application Support/PrismLauncher/instances/26.1.2(1)/minecraft/crash-reports/crash-2026-04-25_18.23.59-client.txt
Process exited with code 1.
Clipboard copy at: 25 Apr 2026 18:26:41 +0200
```

--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -12,7 +12,7 @@ fabric-loader-version = "0.19.2"
 fabric-api-version = "0.145.4+26.1.2"
 forge-version = "26.1.2-64.0.4"
 forgeEventbusValidator-version = "7.0.5"
-neoforgeVersion = "26.1.2.1-beta"
+neoforgeVersion = "26.1.2.21-beta"
 
 # Mod dependencies
 modmenu-version = "18.0.0-alpha.8"

--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -12,7 +12,7 @@ fabric-loader-version = "0.19.2"
 fabric-api-version = "0.145.4+26.1.2"
 forge-version = "26.1.2-64.0.5"
 forgeEventbusValidator-version = "7.0.5"
-neoforgeVersion = "26.1.2.21-beta"
+neoforgeVersion = "26.1.2.29-beta"
 
 # Mod dependencies
 modmenu-version = "18.0.0-alpha.8"

--- a/neoforge/src/main/java/fr/rakambda/fallingtree/neoforge/client/cloth/ClothConfigHook.java
+++ b/neoforge/src/main/java/fr/rakambda/fallingtree/neoforge/client/cloth/ClothConfigHook.java
@@ -20,8 +20,6 @@ import fr.rakambda.fallingtree.common.config.real.cloth.ClothHookBase;
 import fr.rakambda.fallingtree.common.wrapper.IComponent;
 import me.shedaniel.clothconfig2.api.ConfigBuilder;
 import net.minecraft.network.chat.Component;
-import net.neoforged.api.distmarker.Dist;
-import net.neoforged.api.distmarker.OnlyIn;
 import net.neoforged.fml.ModContainer;
 import net.neoforged.neoforge.client.gui.IConfigScreenFactory;
 import org.jspecify.annotations.NonNull;
@@ -52,7 +50,6 @@ public class ClothConfigHook extends ClothHookBase{
 		});
 	}
 	
-	@OnlyIn(Dist.CLIENT)
 	public void fillConfigScreen(@NonNull ConfigBuilder builder, @NonNull Configuration config){
 		var reverseSneakingEntry = builder.entryBuilder()
 				.startEnumSelector(translatable(getFieldName(null, "sneakMode")), SneakMode.class, config.getSneakMode())
@@ -91,7 +88,6 @@ public class ClothConfigHook extends ClothHookBase{
 		fillEnchantmentConfigScreen(builder, config.getEnchantment());
 	}
 	
-	@OnlyIn(Dist.CLIENT)
 	private void fillTreesConfigScreen(@NonNull ConfigBuilder builder, @NonNull TreeConfiguration config){
 		var breakModeEntry = builder.entryBuilder()
 				.startEnumSelector(translatable(getFieldName("trees", "breakMode")), BreakMode.class, config.getBreakMode())
@@ -294,7 +290,6 @@ public class ClothConfigHook extends ClothHookBase{
 		trees.addEntry(trunkLootPercentageEntry);
 	}
 	
-	@OnlyIn(Dist.CLIENT)
 	private void fillToolsConfigScreen(@NonNull ConfigBuilder builder, @NonNull ToolConfiguration config){
 		var ignoreToolsEntry = builder.entryBuilder()
 				.startBooleanToggle(translatable(getFieldName("tools", "ignoreTools")), config.isIgnoreTools())
@@ -362,7 +357,6 @@ public class ClothConfigHook extends ClothHookBase{
 		tools.addEntry(forceToolUsageEntry);
 	}
 	
-	@OnlyIn(Dist.CLIENT)
 	private void fillPlayerConfigScreen(@NonNull ConfigBuilder builder, @NonNull PlayerConfiguration config){
 		var allowedTagsEntry = builder.entryBuilder()
 				.startStrList(translatable(getFieldName("player", "allowedTags")), config.getAllowedTags())
@@ -375,7 +369,6 @@ public class ClothConfigHook extends ClothHookBase{
 		tools.addEntry(allowedTagsEntry);
 	}
 	
-	@OnlyIn(Dist.CLIENT)
 	private void fillEnchantmentConfigScreen(@NonNull ConfigBuilder builder, @NonNull EnchantmentConfiguration config){
 		var requireEnchantmentEntry = builder.entryBuilder()
 				.startBooleanToggle(translatable(getFieldName("enchantment", "requireEnchantment")), config.isRequireEnchantment())

--- a/neoforge/src/main/java/fr/rakambda/fallingtree/neoforge/common/FallingTreeCommonsImpl.java
+++ b/neoforge/src/main/java/fr/rakambda/fallingtree/neoforge/common/FallingTreeCommonsImpl.java
@@ -46,6 +46,7 @@ import net.minecraft.world.level.block.state.BlockState;
 import net.neoforged.bus.api.IEventBus;
 import net.neoforged.neoforge.common.NeoForge;
 import net.neoforged.neoforge.event.level.BlockEvent;
+import net.neoforged.neoforge.event.level.block.BreakBlockEvent;
 import org.jspecify.annotations.NonNull;
 import java.util.Arrays;
 import java.util.HashMap;
@@ -68,7 +69,7 @@ public class FallingTreeCommonsImpl extends FallingTreeCommon<Direction>{
 	private final TagKey<Enchantment> chopperEnchantmentTag;
 	@Getter
 	private final Map<BreakMode, TagKey<Enchantment>> breakModeChopperEnchantmentTag;
-	private final List<BlockEvent.BreakEvent> breakEvents;
+	private final List<BreakBlockEvent> breakEvents;
 	
 	private final Map<IBlock, Boolean> isLogBlockCache;
 	private final Map<IBlock, Boolean> isLeafBlockCache;
@@ -226,9 +227,9 @@ public class FallingTreeCommonsImpl extends FallingTreeCommon<Direction>{
 	
 	@Override
 	public boolean isOwnEvent(@NonNull IBlockBreakEvent event){
-		var result = breakEvents.contains((BlockEvent.BreakEvent) event.getRaw()) || event instanceof FallingTreeBlockBreakEvent;
+		var result = breakEvents.contains((BreakBlockEvent) event.getRaw()) || event instanceof FallingTreeBlockBreakEvent;
 		if(result){
-			breakEvents.remove((BlockEvent.BreakEvent) event.getRaw());
+			breakEvents.remove((BreakBlockEvent) event.getRaw());
 		}
 		return result;
 	}

--- a/neoforge/src/main/java/fr/rakambda/fallingtree/neoforge/common/wrapper/BlockBreakEventWrapper.java
+++ b/neoforge/src/main/java/fr/rakambda/fallingtree/neoforge/common/wrapper/BlockBreakEventWrapper.java
@@ -5,14 +5,14 @@ import fr.rakambda.fallingtree.common.wrapper.IBlockPos;
 import lombok.Getter;
 import lombok.RequiredArgsConstructor;
 import lombok.ToString;
-import net.neoforged.neoforge.event.level.BlockEvent;
+import net.neoforged.neoforge.event.level.block.BreakBlockEvent;
 import org.jspecify.annotations.NonNull;
 
 @RequiredArgsConstructor
 @ToString
 public class BlockBreakEventWrapper implements IBlockBreakEvent{
 	@Getter
-	private final BlockEvent.@NonNull BreakEvent raw;
+	private final @NonNull BreakBlockEvent raw;
 	
 	@Override
 	@NonNull

--- a/neoforge/src/main/java/fr/rakambda/fallingtree/neoforge/event/BlockBreakListener.java
+++ b/neoforge/src/main/java/fr/rakambda/fallingtree/neoforge/event/BlockBreakListener.java
@@ -12,6 +12,7 @@ import net.minecraft.server.level.ServerLevel;
 import net.neoforged.bus.api.SubscribeEvent;
 import net.neoforged.neoforge.event.entity.player.PlayerEvent;
 import net.neoforged.neoforge.event.level.BlockEvent;
+import net.neoforged.neoforge.event.level.block.BreakBlockEvent;
 import org.jspecify.annotations.NonNull;
 import javax.annotation.Nonnull;
 
@@ -44,7 +45,7 @@ public class BlockBreakListener{
 	}
 	
 	@SubscribeEvent
-	public void onBlockBreakEvent(@Nonnull BlockEvent.BreakEvent event){
+	public void onBlockBreakEvent(@Nonnull BreakBlockEvent event){
 		if(event.isCanceled()){
 			return;
 		}

--- a/neoforge/src/main/java/fr/rakambda/fallingtree/neoforge/event/FallingTreeBlockBreakEvent.java
+++ b/neoforge/src/main/java/fr/rakambda/fallingtree/neoforge/event/FallingTreeBlockBreakEvent.java
@@ -5,8 +5,9 @@ import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.block.state.BlockState;
 import net.neoforged.neoforge.event.level.BlockEvent;
+import net.neoforged.neoforge.event.level.block.BreakBlockEvent;
 
-public class FallingTreeBlockBreakEvent extends BlockEvent.BreakEvent{
+public class FallingTreeBlockBreakEvent extends BreakBlockEvent{
 	public FallingTreeBlockBreakEvent(Level level, BlockPos pos, BlockState state, Player player){
 		super(level, pos, state, player);
 	}

--- a/neoforge/src/main/resources/META-INF/neoforge.mods.toml
+++ b/neoforge/src/main/resources/META-INF/neoforge.mods.toml
@@ -15,7 +15,7 @@ file="META-INF/neoforge.accesstransformer.cfg"
 [[dependencies.fallingtree]]
 modId="neoforge"
 type="required"
-versionRange="[26.1.2,)"
+versionRange="[26.1.2.21,)"
 ordering="NONE"
 side="BOTH"
 [[dependencies.fallingtree]]

